### ISSUE TITLE
[lldb-vscode] Delayed text output to the terminal (Windows)

### ISSUE THREAD

**Issue by d-mozulyov** [llvm/llvm-project#62338]
### Problem

The lldb-vscode debugger on Windows launches the application in an external window (read more [here](https://github.com/llvm/llvm-project/issues/62336)). The problem is that when debugging, the terminal state is not updated - the text appears with a huge delay. At the same time, the gdb and CodeLLDB debuggers work correctly.

The second problem is that the text is displayed in the DEBUG CONSOLE tab and how it happens. After printf(), nothing appears, but after cout, everything is displayed. But I would like to point out that duplicating the terminal stream in DEBUG CONSOLE is pointless and confusing. Most likely there we need to display important messages of the LLDB itself. Maximum - cerr messages.

![empty_screen](https://user-images.githubusercontent.com/10961593/234129604-780644ea-7df6-49e3-a1db-9f07c7706b98.png)

### Build lldb-vscode for Windows

1. Download [lldb-vscode-bug.zip](https://github.com/llvm/llvm-project/files/11314121/lldb-vscode-bug.zip) and extract it to a local directory, for example, to `C:\LLDBfix`
2. Clone LLVM 16.0.2 into the sources folder by running the `clone.ps1` script (preferably using a modern version of [PowerShell](https://learn.microsoft.com/en-us/powershell/scripting/install/installing-powershell-on-windows?view=powershell-7.3))
3. Install an up-to-date C++ compiler for Windows (for example, you can unpack the [winlibs archive](https://github.com/brechtsanders/winlibs_mingw/releases/download/12.2.0-16.0.0-10.0.0-ucrt-r5/winlibs-x86_64-posix-seh-gcc-12.2.0-llvm-16.0.0-mingw-w64ucrt-10.0.0-r5.zip) and add the path to the `bin` directory to `PATH`)
4. Build lldb-vscode for Windows by running the `build-lldb.ps1` script

### Reproducing the problem

1. Install Node.js modules into the local directory (for example, to `C:\LLDBfix`): `npm install `
2. Run VSCode in the directory
3. Run "Extension" launch
4. Set a breakpoint on line 7 of `sample.cpp`
5. Run "Debug" launch in target sample project

**Comment by rupprecht** [comment]
The lack of terminal output initially makes sense to me, due to stdout buffering. For a buffered stream, I would expect no output until it's flushed -- the OS might decide to do that, or the program can explicitly flush it. Using `std::endl` includes a flush: see https://en.cppreference.com/w/cpp/io/manip/endl.

However, according to https://manpages.debian.org/bullseye/manpages-dev/stdout.3.en.html#NOTES, it's line-buffered when connected to a terminal. That could be an explanation for what you see: maybe GDB and CodeLLDB both configure the subprocesses as terminals, in which case you see output anytime there's a `\n` char. I think LLDB is doing the same, but maybe not in the Windows + LLDB VSCode path. (Personally, I can't yet reproduce this issue -- I see line buffering behavior)

Do you see this problem go away if you use stderr for the `printf` command? i.e. `fprintf(stderr, "Hello from printf\n");

**Comment by d-mozulyov** [comment]
@rupprecht 

Can you please tell me if you use the reproduce steps that I described? The issue only reproduces on Windows and only in lldb-vscode.

**Comment by rupprecht** [comment]
Yes, I wasn't able to reproduce because I don't have access to a Windows machine. Following the same steps but on Linux don't reproduce. Hopefully someone with Windows experience can take a look at this.

**Comment by d-mozulyov** [comment]
@rupprecht 

If you wish, I can provide RDP access to our build server on Windows. There you can reproduce the error and try to fix it. If you want, please contact me via Telegram @dmozulyov.

**Comment by d-mozulyov** [comment]
@rupprecht 

I also made significant changes to the description of the issue, you can reproduce the more important problem on Linux or macOS.
